azimuth/startup.py

```diff
@@ -146,16 +183,22 @@ def make_startup_tasks(
     dms = {k: v for k, v in dataset_split_managers.items() if v is not None}
     tasks: Dict[DatasetSplitName, DaskModule] = {}
 
-    for dataset_split_name, dm in dms.items():
+    if DatasetSplitName.all not in dataset_split_names:
+        available_datasets_splits = list(set(dataset_split_names) & set(dms))
+    else:
+        available_datasets_splits = [DatasetSplitName.all]
```

Comment:
Similarly, here I would remove the `not` and flip. Also, there is no `s` to `datasets`.
```suggestion
    if DatasetSplitName.all in dataset_split_names:
        available_dataset_splits = [DatasetSplitName.all]
    else:
        available_dataset_splits = list(set(dataset_split_names) & set(dms))
```
Alternatively though, here is another way with less list->set and set->list conversions:
```suggestion
    available_dataset_splits = [n for n in dataset_split_names if n in dms or n == DatasetSplitName.all]
```